symposium/macos-app/Sources/Symposium/Views/ProjectView.swift

```diff
@@ -765,10 +759,16 @@ struct DeleteTaskspaceDialog: View {
             }
         }
         .onAppear {
-            // Set default deleteBranch toggle based on safety analysis
-            // Safe branches (no unmerged commits, no uncommitted changes): checked by default (encourage cleanup)
-            // Risky branches: unchecked by default (prevent accidental loss)
-            deleteBranch = (branchInfo.unmergedCommits == 0 && !branchInfo.hasUncommittedChanges)
+            Task {
+                let manager = projectManager
+                let ts = taskspace
+                cachedBranchInfo = await Task.detached {
+                    manager.getTaskspaceBranchInfo(for: ts)
+                }.value
+                
+                // Set default deleteBranch toggle based on safety analysis
+                deleteBranch = (cachedBranchInfo.unmergedCommits == 0 && !cachedBranchInfo.hasUncommittedChanges)
```

Comment:
I guess what happens is that, initially, the cached branch name is empty, so the user sees no checkbox at all, but then it appears some time later once the git command has completed? I think it'd be nice if we displayed a message like "...determining github branch information..." to keep them up-to-date.